File: LegacySource/settlement/room/main/util/RoomInit.cs
using System;
using System.Collections.Generic;

namespace Settlement.Room.Main.Util
{
    public sealed class RoomInit
    {
        public readonly double[] Res;
        public readonly double[] Stats;
        public double ResMul = 1;
        public readonly int Degrade;

        public RoomInit(RoomBlueprintImp b, int degrade)
        {
            if (b.Constructor() != null)
            {
                Res = new double[b.Constructor().Resources()];
                Stats = new double[b.Constructor().Stats().Count];
            }
            else
            {
                Stats = new double[0];
                Res = new double[0];
            }
            this.Degrade = degrade;
        }
    }
}

File: settlement/room/main/util/RoomInit.java
package settlement.room.main.util;

import settlement.room.main.RoomBlueprintImp;

public final class RoomInit {

	public final double[] res;
	public final double[] stats;
	public double resMul = 1;
	public final int degrade;
	
	public RoomInit(RoomBlueprintImp b, int degrade) {
		if (b.constructor() != null) {
			res = new double[b.constructor().resources()];
			stats = new double[b.constructor().stats().size()];
		}
		else {
			stats = new double[0];
			res = new double[0];
		}
		this.degrade = degrade;
	}
	
}
